### PR TITLE
feat(FC): let's write this as a FunctionComponent in Vue3

### PR DESCRIPTION
**Pull request by little-buddy**
Composition API to write component

```js
import {defineComponent} from "vue"

export default defineComponent({
  setup(){
    // logic

    return () => <Component />;
  }
})
```

Then
```
app.config.globalProperties.$xxx = xxx

change: need provide in root

  SET => provide(key,$xxx)
  GET => inject(key)
```

Some libraries have defined out of the box usage
```
el.
  useRouter、useRoute from vue-router

  useStore from vuex
```

### CODE REVIEW

**Comment by little-buddy** [comment]
Previous $xxx need to passed through the parent element by provide/inject exlcude vue3api
el $watch can exported from vue

File: src/hooks/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  provide($WARNING, Modal.warning)
+  provide($CONFIRM, Modal.confirm)
+  provide($NOTIFICATION, notification)
+  provide($DESTROYALL, Modal.destroyAll)
+  provide($MESSAGE, Message)
+}
+
+export { useStore, useRoute, useRouter }
+export const useError = () => inject($ERROR)
+export const useInfo = () => inject($ERROR)
```
Comment (Linkontoask) [inline src/hooks/index.tsx]: Pay attention here

**Review by Linkontoask — CHANGES_REQUESTED** [review]
This writing method is very good, and I will improve the functions that have been completed before. We can specify a rule (eslint) that does not allow the use of other configuration items in the component, or we can use the following write rules to avoid checking
```
/* eslint-disable no-alert, no-console */

alert('foo');
console.log('bar');

/* eslint-enable no-alert, no-console */
```

There is a small problem in the merge you submitted, remember to fix it.

Oh, by the way, this way of writing can largely avoid the forced conversion of any type, and can make better use of typescript to help us avoid errors

**Comment by Linkontoask** [comment]
I found some problems. Maybe the official vue-router hasn't solved it. I can't think of getting the latest route in the setup. https://next.router.vuejs.org/api/#useroute. I am wondering whether I should use the render option when encountering such things, or use the render option of vuex and the root component to solve

**Review by Linkontoask — COMMENTED** [review]
perfect

**Pull request merged**